source/extensions/network/dns_resolver/cares/dns_impl.cc

```diff
@@ -320,7 +321,7 @@ DnsResolverImpl::AddrInfoPendingResolution::AddrInfoPendingResolution(
     DnsResolverImpl& parent, ResolveCb callback, Event::Dispatcher& dispatcher,
     ares_channel channel, const std::string& dns_name, DnsLookupFamily dns_lookup_family)
     : PendingResolution(parent, callback, dispatcher, channel, dns_name),
-      dns_lookup_family_(dns_lookup_family) {
+      dns_lookup_family_(dns_lookup_family), available_interfaces_(availableInterfaces()) {
```

Comment:
every query will probe the system to check for interfaces. Is this reasonable, or is it the case that the available interfaces should be more stable?

Reply:
Available interfaces as well as whether the interface has an IPv4/IPv6 address can absolutely change over time.

Reply:
I don't see the harm in doing the check each time, since we also have other controls on the frequency of DNS lookups.

Reply:
`getifaddrs()` isn't very cheap (it involves a syscall and a lock in the kernel, but worth measuring though) so it would make sense to cache the value and update when the OS tells you things change, but that's prone to races so I guess calling `getifaddrs()` for each resolution is reasonable